pkg/liqo-controller-manager/storageprovisioner/remoteprovisioner.go

```diff
@@ -13,3 +13,145 @@
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"sigs.k8s.io/sig-storage-lib-external-provisioner/v7/controller"
+
+	"github.com/liqotech/liqo/pkg/virtualKubelet/forge"
+)
+
+// ProvisionRemotePVC ensures the existence of a remote PVC and returns a virtual PV for that remote storage device.
+func ProvisionRemotePVC(ctx context.Context,
+	options controller.ProvisionOptions,
+	remoteNamespace, remoteRealStorageClass string,
+	remotePvcLister corev1listers.PersistentVolumeClaimNamespaceLister,
+	remotePvcClient corev1clients.PersistentVolumeClaimInterface) (*v1.PersistentVolume, controller.ProvisioningState, error) {
+	virtualPvc := options.PVC
+	if _, err := createOrUpdate(ctx, remotePvcLister, remotePvcClient, virtualPvc.Name,
+		func(pvc *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaim, error) {
+			return mutateRemoteRealPVC(virtualPvc, pvc, remoteRealStorageClass, remoteNamespace)
+		}); err != nil {
+		return nil, controller.ProvisioningInBackground, err
+	}
+
+	pv := &v1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: options.PVName,
+		},
+		Spec: v1.PersistentVolumeSpec{
+			StorageClassName: options.StorageClass.Name,
+			AccessModes:      options.PVC.Spec.AccessModes,
+			Capacity: v1.ResourceList{
+				v1.ResourceStorage: options.PVC.Spec.Resources.Requests[v1.ResourceStorage],
+			},
+			PersistentVolumeSource: v1.PersistentVolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: "/tmp/liqo-placeholder",
+				},
+			},
+			NodeAffinity: &v1.VolumeNodeAffinity{
+				Required: &v1.NodeSelector{
+					NodeSelectorTerms: []v1.NodeSelectorTerm{
+						{
+							MatchExpressions: []v1.NodeSelectorRequirement{
+								{
+									Key:      "kubernetes.io/hostname",
+									Operator: v1.NodeSelectorOpIn,
+									Values:   []string{options.SelectedNode.Name},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if options.StorageClass.ReclaimPolicy != nil {
+		pv.Spec.PersistentVolumeReclaimPolicy = *options.StorageClass.ReclaimPolicy
+	}
+
+	return pv, controller.ProvisioningFinished, nil
+}
+
+// mutateRemoteRealPVC enforces the remote realPVC fields merging its old values (if any)
+// with the ones coming from the virtualPVC.
+// i.e. the PVC spec is the copy of the virtual one, but the storage class is the one set in the
+// storage provisioner or the one previously set in the PVC (since it is a read-only field). The
+// real volumeName is preserved too.
+func mutateRemoteRealPVC(virtualPvc, realPvc *v1.PersistentVolumeClaim,
+	storageClass, namespace string) (*v1.PersistentVolumeClaim, error) {
+	if realPvc == nil {
+		realPvc = &v1.PersistentVolumeClaim{
+			TypeMeta: virtualPvc.TypeMeta,
+		}
+	}
+
+	realPvc.ObjectMeta.Name = virtualPvc.Name
+	realPvc.ObjectMeta.Namespace = namespace
+
+	if realPvc.ObjectMeta.Labels == nil {
+		realPvc.ObjectMeta.Labels = map[string]string{}
+	}
+
+	for k, v := range forge.ReflectionLabels() {
+		realPvc.ObjectMeta.Labels[k] = v
+	}
```

Comment:
Could use labels.Merge. At the moment, local PVC labels are not reflected (if that makes sense).